src/test/rpc/NoRipple_test.cpp

```diff
@@ -0,0 +1,226 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/test/jtx.h>
+
+namespace ripple {
+
+namespace test {
+
+class NoRipple_test : public beast::unit_test::suite
+{
+public:
+    void
+    testSetAndClear()
+    {
+        testcase("Set and clear noripple");
+
+        using namespace jtx;
+        Env env(*this);
+
+        auto const gw = Account("gateway");
+        auto const alice = Account("alice");
+
+        env.fund(XRP(10000), gw, alice);
+
+        auto const USD = gw["USD"];
+
+        Json::Value account_gw;
+        account_gw[jss::account] = gw.human();
+        Json::Value account_alice;
+        account_alice[jss::account] = alice.human();
+
+        for (auto SetOrClear : {true,false})
+        {
+            // Create a trust line with no-ripple flag setting
+            env( trust(gw, USD(100), alice, SetOrClear ? tfSetNoRipple
+                                                       : tfClearNoRipple));
+            env.close();
+
+            // Check no-ripple flag on sender 'gateway'
+            auto lines = env.rpc("json", "account_lines", to_string(account_gw));
+            auto const& gline0 = lines[jss::result][jss::lines][0u];
+            BEAST_EXPECT(gline0[jss::no_ripple].asBool() == SetOrClear);
+
+            // Check no-ripple peer flag on destination 'alice'
+            lines = env.rpc("json", "account_lines", to_string(account_alice));
+            auto const& aline0 = lines[jss::result][jss::lines][0u];
+            BEAST_EXPECT(aline0[jss::no_ripple_peer].asBool() == SetOrClear);
```

Comment:
I am confused... we set `no ripple` on one side of the trust line, but we look for it in both?